src/java/org/apache/cassandra/service/accord/serializers/RecoverySerializers.java

```diff
@@ -86,8 +86,12 @@ public void serialize(RecoverReply reply, DataOutputPlus out, int version) throw
                 CommandSerializers.deps.serialize(recoverOk.earlierCommittedWitness, out, version);
                 CommandSerializers.deps.serialize(recoverOk.earlierAcceptedNoWitness, out, version);
                 out.writeBoolean(recoverOk.rejectsFastPath);
-                CommandSerializers.writes.serialize(recoverOk.writes, out, version);
-                AccordData.serializer.serialize((AccordData) recoverOk.result, out, version);
+                out.writeBoolean(recoverOk.writes != null);
```

Comment:
I didn't look closer into all reasons writes and result are null, but did see that there were cases (such as recovery), so adding a bool up from lets me check.

One optimization I didn't do is that we could have a single byte handle all null checks by just using each bit for a field...